### PR TITLE
use IO ports on bare Linux

### PR DESCRIPTION
**Pull request by jklmnn**
This enables IO Ports to be used for Genode running on a bare Linux kernel.
To test this changes, build from 80d9ec9b639e59b36dd8742dc13f88a40c03fc9a which contains further changes to run the test (disable the rtc driver to use its own implementation if run with spec linux, custom runfile to run this test (run/rtc)).

### CODE REVIEW

File: repos/base-linux/src/core/io_port_session_component.cc [hunk 1]
```diff
@@ -0,0 +1,19 @@
+
+#include <util/string.h>
+#include <util/arg_string.h>
+#include <root/root.h>
+#include <io_port_session_component.h>
+
+
+Genode::Io_port_session_component::Io_port_session_component(Genode::Range_allocator *io_port_alloc,
+                                                     const char      *args)
```
Comment (senier) [inline repos/base-linux/src/core/io_port_session_component.cc]: Is that alignement issue a Github problem or in the code?

File: repos/base-linux/src/core/io_port_session_component.cc [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+: _io_port_alloc(io_port_alloc)
+{
+    //FIXME: throw service denied if we're not root (or init)!
```
Comment (senier) [inline repos/base-linux/src/core/io_port_session_component.cc]: Not here, rather deny the session...

File: repos/base-linux/src/core/platform_services.cc [hunk 3]
```diff
@@ -0,0 +1,38 @@
+/*
+ * \brief  Platform specific services for x86
+ * \author Stefan Kalkowski
+ * \date   2012-10-26
```
Comment (senier) [inline repos/base-linux/src/core/platform_services.cc]: Change description, add yourself as author and update date.

File: repos/base-linux/src/core/platform_services.cc [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+ */
+
+/*
+ * Copyright (C) 2012-2017 Genode Labs GmbH
```
Comment (senier) [inline repos/base-linux/src/core/platform_services.cc]: Update.

File: repos/base-linux/src/core/platform_services.cc [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+#include <core_linux_syscalls.h>
+
+/*
+ * Add x86 specific ioport service
+ */
+void Genode::platform_add_local_services(Rpc_entrypoint*,
+                                         Sliced_heap *sliced_heap,
+                                         Registry<Service> *local_services)
+{
+        lx_iopl(3);
```
Comment (senier) [inline repos/base-linux/src/core/platform_services.cc]: As discussed f2f today: check iopl() return value and execute below code only on success. The session should be denied in this case.

**Review by senier — CHANGES_REQUESTED** [review]

**Review by senier — COMMENTED** [review]
This looks OK now. One more style issue: Please review your changes to consistently use spaces for indentation. (I hope you know that this is important: https://stackoverflow.blog/2017/06/15/developers-use-spaces-make-money-use-tabs/ ;-)

**Comment by jklmnn** [comment]
The problem with indentation in Genode is that the Genode style guide proposes tabs. When modifying code from Genode I usually have a mixed indentation (tabs from upstream, spaces for my code) and I convert everythig to tabs before creating a pull request (with `tool/beautify`).
So I'm currently not 100% sure which indentation scheme I should use (keeping it mixed is inconsistentm converting everything to spaces requires double work and converting everything to tabs breaks my workflow).

**Comment by senier** [comment]
I see. That's a bit annoying indeed. I lean towards fixing the whitespace issue when we merge into our own topic branch (the one that we later use for a pull request towards Genode). Rationale: This increases the chance that our commits get accepted unmodified which could ease rebasing onto upstream later.

**Comment by jklmnn** [comment]
As we have discussed we fix the indentation when getting the changes upstream. Was there anything to do yet beside this? If not I'd ask you to approve the review.

Beside that you asked me to review my changes but unfortunately I can't accept them.

pr

**Review by senier — APPROVED** [review]

**Pull request merged**